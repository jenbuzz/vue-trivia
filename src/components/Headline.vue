<template>
    <b-row align-h="center">
        <b-col sm="6">
            <h2 class="text-center mb-3" role="button" @click="reset">Trivia Quiz</h2>
        </b-col>
    </b-row>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Headline extends Vue {
    @Prop() private reset!: Function;
}
</script>
