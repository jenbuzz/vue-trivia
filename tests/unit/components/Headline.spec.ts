import { createLocalVue, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Headline from '@/components/Headline.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Component: Headline', () => {
    it('renders', () => {
        const wrapper = shallowMount(Headline, {
            localVue,
            propsData: { reset: jest.fn() },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
