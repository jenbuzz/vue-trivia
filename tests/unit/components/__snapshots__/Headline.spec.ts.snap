// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Headline renders 1`] = `
<b-row-stub tag="div" alignh="center">
  <b-col-stub sm="6" tag="div">
    <h2 role="button" class="text-center mb-3">Trivia Quiz</h2>
  </b-col-stub>
</b-row-stub>
`;
